src/service/url-impl.js

```diff
@@ -141,6 +139,26 @@ export class Url {
   getWinOrigin(win) {
     return win.origin || this.parse(win.location.href).origin;
   }
+
+  /**
+   * If the resource URL is referenced from the publisher's origin,
+   * convert the URL to be referenced from the cache.
+   * @param {string} resourceUrl The URL of the document to load
+   * @return {string}
+   */
+  getCdnUrlOnOrigin(resourceUrl) {
+    if (isProxyOrigin(resourceUrl)) {
+      return resourceUrl;
+    }
+
+    const {
+      host,
+      hash,
+      pathname,
+      search,
+    } = this.parse(resourceUrl);
+    return `${urls.cdn}/c/${host}${pathname}${search}${hash}`;
```

Comment:
I think we need to `encodeURIComponent` `host` as parse will be retuning decoded version.

Reply:
Huh?

Reply:
I thought `parse` returns unicode of the IDN hostname, looks like it returns Punycode so I guess there is no need to encode the host in that case but I am not sure if cache expects punycode as host after the `/c/`. Testing `http://δοκιμή.com` with https://ampbyexample.com/advanced/using_the_google_amp_cache/, looks like punycode is used for subdomain name but the hostname after `/c/` is Uri encoded IDN. @honeybadgerdontcare would cache Url work if host after `/c/` is punycode?

Reply:
chances of an IDN using live-list and SxG is pretty much 0 at the moment, so don't block on this comment.

Reply:
Sorry for the delay. I'll find out and respond here.